l2geth/internal/ethapi/api.go

```diff
@@ -1135,6 +1136,18 @@ func legacyDoEstimateGas(ctx context.Context, b Backend, args CallArgs, blockNrO
 			return 0, fmt.Errorf("gas required exceeds allowance (%d)", cap)
 		}
 	}
+	//gas_required = (SIMULATE_GAS_ESTIMATE + CONSTANT_OVERHEAD + VARIABLE_OVERHEAD * CALLDATA_SIZE) * (64/63)^UNACCOUNTED_CALL_DEPTH
+
+	// add the constant overhead
+	hi += 200_000
+	// add variable overhead
+	if args.Data != nil {
+		hi += uint64(len(*args.Data)) * 20
+	}
+	// add unaccounted call depth
+	callDepth := gmath.Pow(64/63, 6)
```

Comment:
@tynes reminder that we realized this overhead calculation was wrong

Reply:
We can get rid of this now because our gasLimit is actually used to made the **call** in the wallet contract. That means that if you do a normal `simulateMessage` it should actually supply an upper bound for the gas limit.

Note this does not fundamentally solve our issue where the block gas limit that users have available to them is `MAX_GAS_LIMIT - EM overhead - sequencer entrypoint overhead - call depth` that you were calculating here. However, the good news is that at least users don't need to sign off on that overhead & can instead focus on how much gas they personally have been supplied.

So the TLDR here is: we do not need to return a gas limit which takes into consideration this gasLimit logic.  Instead this logic should be used to detect whether the transaction will be **unreachable** because `gasLimit + overheadToReachTheTxFromSequencerEntrypoint > blockGasLimit`